### PR TITLE
Allows setting of mute params during initialization

### PR DESCRIPTION
**Pull request by robgonnella**

### CODE REVIEW

**Comment by CapOM** [comment]
What is the difference between mute audio and mute video ?  (audio only VS video+audio streams ? )
Do we need something similar to have the camera off from initialization too ? For example if I turn off camera in rumpus before joining a meeting I can still still the green light of the camera for a few seconds (it lights up and turn off automatically after 2 sec after joining the meeting, if off before joining)

Thx!

**Comment by robgonnella** [comment]
Prior to this change we were waiting for successful connection then appropriately toggling video and/or audio based on our local state. This is why you would see green light for a few seconds before it finally got toggled off. With this change we can actually pass these parameters in at initialization. You may still see a split second of a green light, but it should be much shorter and you really shouldn't ever see the actual stream.

**Comment by CapOM** [comment]
Ah ok so muteVideo means turn off camera ?

**Comment by robgonnella** [comment]
Ahh just found another bug though. Start app, mute video, start meeting -- video is muted ---
unmute video -- nothing happens --- toggle video off, toggle video on again --- stream appears --- mute video --- stream disappears but green light stays on

**Comment by robgonnella** [comment]
Yes "muteVideo" means turn off camera. Sorry I realize now that might be misleading, I can change to muteCamera and muteMicrophone if you want. Also just pushed up again with a fix for the bug I mentioned above

File: src/rtc-client.js [hunk 1]
```diff
@@ -65,10 +67,19 @@ module.exports = function (RTCManager) {
...
     remoteVideoEl = options.remoteVideoEl;
     contentVideoEl = options.contentVideoEl;
+    if (typeof options.muteAudio === 'boolean') {
+      config.muteParams.localAudio = options.muteAudio;
+      setMutes = true;
+    }
+    if (typeof options.muteVideo === 'boolean') {
+      config.muteParams.localVideo = options.muteVideo;
+      setMutes = true;
+    }
```
Comment (CapOM) [inline src/rtc-client.js]: The name muteVideo is fine, thx for the clarification. Just another question, when setting muteAudio it does mute the video too as they toggle the same variable `setMutes` ?

Comment (robgonnella) [reply src/rtc-client.js]: I was conflicted about this too. The way the api is set up it handles both audio and video in one function. I was thinking about splitting them but I was afraid of pulling apart the api too much. However I just found another bug which is similar to the one I described above.

launch app, make sure video is on, start meeting, leave meeting, rejoin meeting -- video is still on as it should be --- toggle video off --- nothing happens -- toggle video on --- nothing happens -- toggle video off -- video is muted but green light stays on

Going to take one more crack at fixing this

Comment (CapOM) [reply src/rtc-client.js]: Ah ok thx, I miss read, the argument of RTCManager.muteStreams is config.muteParams, not setMutes.

**Comment by robgonnella** [comment]
Ok landed at something that mostly works. Theres still one bug that I can't seem to figure out but maybe we can tackle in a separate patch as this is still an improvement.

Bug I can't seem to figure out is as follows:

• Launch app and mute audio but leave video unmuted
• Start meeting
• Once meeting starts mute & unmute video (everything works as expected)
• Leave meeting
• Rejoin meeting (again audio muted and video unmuted)
• Once meeting is started mute video (nothing happens)
• Try to again "unmute" video (slight flash but nothing happens)
• Try one more time - This time it works and states are aligned again

**Comment by CapOM** [comment]
Looks good and I agree that we can take care of other corner cases later. Feel free to merge!

**Comment by robgonnella** [comment]
Thanks for the review Julien!

**Pull request merged**